Spigot-Server-Patches/0381-Optimize-redstone-algorithm.patch

```diff
@@ -0,0 +1,1185 @@
+From 6a5104193614fbe321dca74d32646a4ad33595fa Mon Sep 17 00:00:00 2001
+From: egg82 <phantom_zero@ymail.com>
```

Comment:
I'd say it's worth using @theosib here - he wrote 99% of this.

Reply:
I credited the original author multiple times. Although a LOT of the work is his I still had to learn how it worked and get it to play nicely with Paper.

Reply:
You can always add yourself as co-author:
https://help.github.com/articles/creating-a-commit-with-multiple-authors/

Reply:
Worked it out in Discord. My intent isn't to steal credit for work. There's no way I'd ever want to do that. The patch is basically theosib's, but it still took me a fair amount of hours, work, and understanding to implement it correctly and get it to play nicely with Paper and Bukkit's events. I just didn't want that to go by without having a small mention.

The solution we collectively came up with was to replace that header with a small "Ported to Paper and updated to 1.13 by egg82" and add GitHub's Co-authored-by header to the patch while attributing the real Author header to theosib

Reply:
I appreciate the effort you put into adapting this code and you should not be shy about getting credit for that contribution.  There are multiple ways to handle this.  You can name me as the author and mention prominently in notes or other metadata that you did a lot of work to port it.  Or you can say that you are the author of THIS VERSION of the accelerator, and mention in appropriate places that you adapted it from my code.  Also, I would suggest adding comments to the code that explain the lineage and revision history.  I don't feel strongly about any of this.  I did this initially just to for an interesting challenge and to improve redstone performance.  But when I showed it to Grum, he of course had to make up all sorts of reasons why it was inadequate, essentially giving me what he thought was an impossible combination of requirements.  He didn't know who he was dealing with.  :)  Anyhow, I would be perfectly happy if Mojang were to absorb this code, and I wouldn't want any attribution requirements to be a stumbling block.  Same goes for Paper.  I'm leaving ethics of proper attribution to you; whatever you come up with, I'm sure I'll be fine with it.

Reply:
Honestly I would feel bad if I tried to claim it as my own. Through reading and understanding your code it's obvious that the brilliance of the thing is in its simplicity, and the only major complexities presented are in keeping with vanilla behavior and actually linking it into the Mojang server. BUT there's still a ton of work that went into the thing so I would be very uncomfortable trying to take credit for it. I wouldn't even call it a new version, so I wouldn't try to claim that either.

I'm honestly more than fine with you taking the vast majority of the credit- I just didn't want my name to be completely lost when I spent a fair portion of my night (and half of my morning as well) actually getting it to work on Paper is all. For me, it's not about trying to take credit. It's just about appropriately crediting the people who worked on it- and since that includes a few tiny contributions from myself, my name as even just a footnote somewhere is more than okay. I kept the comments about other contributors in for exactly the same reason ;)